### PR TITLE
bugfix/portland-supporting-files

### PR DESCRIPTION
**Pull request by dphoria**
### Link to Relevant Issue

This pull request is part of #43 

### Description of Changes

_Include a description of the proposed changes._

`PortlandScraper.get_supporting_files()` fixed to fit with recent changes. The file was also clean up and docstrings were added.

### CODE REVIEW

**Comment by dphoria** [comment]
Oops, want to update `__init__.py` first so we can officially start using this from `cdp-backend` pipeline.

**Comment by dphoria** [comment]
/test-scraper get_portland_events --from_dt=2022-02-01T00:00:00 --to_dt=2022-02-02T00:00:00

**Comment by dphoria** [comment]
Oops forgot about partial

**Comment by dphoria** [comment]
/test-scraper get_portland_events --from_dt=2022-02-01T00:00:00 --to_dt=2022-02-02T00:00:00

**Comment by dphoria** [comment]
Might as well also update `portland-static.json` while I'm at it.

**Comment by dphoria** [comment]
/test-scraper get_portland_events --from_dt=2022-02-01T00:00:00 --to_dt=2022-02-02T00:00:00

**Comment by dphoria** [comment]
Because this is not a fix of an existing deployed file, I think it's probably better/easier to just look at the new files, as opposed to the diff set.
- [`portland.py`](https://github.com/dphoria/cdp-scrapers/blob/bugfix/portland-support-files/cdp_scrapers/instances/portland.py)
- [`portland-static.json`](https://github.com/dphoria/cdp-scrapers/blob/bugfix/portland-support-files/cdp_scrapers/instances/portland-static.json)

I've also run a test pull in [above comment](https://github.com/CouncilDataProject/cdp-scrapers/pull/72#issuecomment-1032188622). The workflow run result is [here](https://github.com/CouncilDataProject/cdp-scrapers/actions/runs/1810267935).

Admittedly the scraper is far from complete. But @Shak and I are trying to get to a place where we can deploy an instance and then continue to improve upon the events scraper.

I'm just going to tag everyone :sweat_smile: any/all thoughts/feedback are welcome!

**Comment by JacksonMaxfield** [comment]
Will give a more thorough review tomorrow. Got a lot of work to finish up tonight but I can already say:

I knew there would come a day when we need to add `youtube-dl` to the library. These videos are hosted on youtube and the embed link does not work for download (and I assume the event page video component).

Time to add a youtube downloader function to `cdp-backend`: https://github.com/ytdl-org/youtube-dl/blob/master/README.md#embedding-youtube-dl

**Comment by dphoria** [comment]
> Will give a more thorough review tomorrow.

Yup, thanks for your input as always. Also, for everyone, this is the referenced event https://www.portland.gov/council/agenda/2022/2/1 for the above test pull.

File: cdp_scrapers/instances/portland.py [hunk 1]
```diff
@@ -48,6 +52,28 @@
...
+MATTER_IN_PROG_PATTERNS = [
+    "passed to",
+    "placed on",
+    "continued",
+    "referred",
+]
+
+MINUTE_ITEM_PASSED_PATTERNS = [
+    "accepted",
+    "passed$",
```
Comment (isaacna) [inline cdp_scrapers/instances/portland.py:70]: Is the `$` intentional?

File: cdp_scrapers/instances/portland.py [hunk 2]
```diff
@@ -100,50 +126,127 @@ def make_efile_url(efile_page_url: str) -> str:
...
+    Notes
+    -----
+    first word in title_and_name is presumed to be title
+    """
+    # title_and_name:
+    #   The title (Mayor of Commissioner) and name of a Portland City Commission
+    #   member.
+    #   e.g., Mayor Ted Wheeler, Commissioner Carmen Rubio
+
+    name_index = title_and_name.find(" ")
```
Comment (isaacna) [inline cdp_scrapers/instances/portland.py:201]: Is there ever a case where the title has a space in it? Like "Vice President"

File: cdp_scrapers/instances/portland.py (hunk 1 above)
Comment (dphoria) [reply cdp_scrapers/instances/portland.py:70]: Yes it is. To differentiate e.g. "passed" and "passed to second reading".

File: cdp_scrapers/instances/portland.py [hunk 3]
```diff
@@ -318,18 +424,27 @@ def get_votes(self, minute_section: Tag) -> Optional[List[Vote]]:
         vote_list = []
         for vote_element in vote_elements:
             vote = vote_element.text.strip()
+            # at this point vote string is like
+            # Commissioner Jo Ann Hardesty Absent
+            # Commissioner Mingus Mapps Yea
+            is_absent = "Absent" in vote
+            vote = vote.replace(" Absent", "")
```
Comment (isaacna) [inline cdp_scrapers/instances/portland.py:431]: When we're doing raw string searches it could be worth normalizing the data first by doing something like `.lower()` on the string. Might not be necessary in this case but something to consider

File: cdp_scrapers/instances/portland.py (hunk 2 above)
Comment (dphoria) [reply cdp_scrapers/instances/portland.py:201]: Ooh great point.. For Portland specifically I don't believe so. I will definitely double check. I _think_ I've only seen either Mayor, or everyone else as Commissioner. Yeah, the terminology is quite different from e.g. Seattle. 

Nonetheless I'm going to make sure on this. It'll obviously break the logic.

File: cdp_scrapers/instances/portland.py (hunk 3 above)
Comment (dphoria) [reply cdp_scrapers/instances/portland.py:431]: Totally agree. Better to be safe than sorry.

File: cdp_scrapers/instances/portland.py [hunk 4]
```diff
@@ -166,14 +269,10 @@ def get_matter(self, minute_section: Tag) -> Optional[Matter]:
...
 
         # Find result status
-        result_status_element_sibling = minute_section.find(
-            "div", text=re.compile("Disposition"), attrs={"class": "field__label"}
-        )
-        result_status_element = result_status_element_sibling.next_sibling
-        result_status = result_status_element.text
-        if result_status in ["Accepted", "Passed"]:
+        result_status = get_disposition(minute_section)
+        if result_status in ["Accepted", "Passed", "Adopted"]:
```
Comment (dphoria) [inline cdp_scrapers/instances/portland.py:273]: I forgot t update this blob. I must have had low sugar in my brain at this time. Will use the `MATTER*PATTERNS` for cleaner code.

**Review by isaacna — APPROVED** [review]
Looks good overall to me! I didn't look through all the scraper logic or compare it with the actual html for the portland web page (I trust you guys lol). But just added some small comments code-wise.

But I did check the scraper results and looked at the json relative to our models and it looked good to me. Also didn't know that we had an action built for that, but that's very useful!

**Comment by dphoria** [comment]
> Looks good overall to me! I didn't look through all the scraper logic or compare it with the actual html for the portland web page (I trust you guys lol). But just added some small comments code-wise.

Thank you so much! :pray: I'm going to go through the actual content comparisons again. I've been known to miss details LOL. The scraper currently is intentionally incomplete, but the pieces we currently get, should be correct.

**Comment by isaacna** [comment]
> Will give a more thorough review tomorrow. Got a lot of work to finish up tonight but I can already say:
> 
> I knew there would come a day when we need to add `youtube-dl` to the library. These videos are hosted on youtube and the embed link does not work for download (and I assume the event page video component).
> 
> Time to add a youtube downloader function to `cdp-backend`: https://github.com/ytdl-org/youtube-dl/blob/master/README.md#embedding-youtube-dl
> 
> Unless @tohuynh or @BrianL3 know some weird embed magic

I may be missing context but is the issue that event video component wouldn't be able to load it? I remember there was this [issue](https://github.com/CouncilDataProject/cdp-backend/issues/129) in cdp-backend but wasn't sure if it was referring to the same thing. Do we need to download the video from YouTube and store it ourselves, or could we just store the original url in the db and fetch the CC file (presumably though some Youtube API)? 

Also I guess we'd need to download every video file to run speaker classification in the future?

**Comment by JacksonMaxfield** [comment]
> I may be missing context but is the issue that event video component wouldn't be able to load it? I remember there was this [issue](https://github.com/CouncilDataProject/cdp-backend/issues/129) in cdp-backend but wasn't sure if it was referring to the same thing. Do we need to download the video from YouTube and store it ourselves, or could we just store the original url in the db and fetch the CC file (presumably though some Youtube API)?
> 
> Also I guess we'd need to download every video file to run speaker classification in the future?

If possible, we should reuse the embed link provided by this scraper on the frontend but not sure if that is possible. I leave that to To and Brian. (This issue _is_ similar but it moreso relates to cross-origin requests I think)

Regardless of above hosting or not, yes we will need to be able to download the video for both audio striping for session content hash generation, speech-to-text usage, _and_ speaker classification.

Basically regardless of hosting strategy, we need to be able to download the video the local process.

**Comment by dphoria** [comment]
/test-scraper get_portland_events --from_dt=2022-02-01T00:00:00 --to_dt=2022-02-02T00:00:00

**Comment by tohuynh** [comment]
> Unless @tohuynh or @BrianL3 know some weird embed magic

video.js documentation says it can play youtube URL, but not out-of-the-box. need to add some sort of plugin I think

**Comment by JacksonMaxfield** [comment]
Sorry for missing this. Just high level question, where does this PR put us with a Portland instance? How close are we basically?
